File: services/stream-processors/base/src/services/clock.py
import os
from datetime import datetime
from typing import Literal

import requests


class Clock:
    @staticmethod
    def now(source: Literal["system", "external"] = "external") -> datetime:
        if source == "system":
            return datetime.now()

        base_url = os.getenv("CLOCK_URL", "http://clock:8000")
        response = requests.get(f"{base_url}/clock?response_type=text")
        return datetime.fromisoformat(response.text)

File: services/classifiers/base-js/src/services/clock.ts
import { env } from '../env.js';

type Source = 'system' | 'external';

export class Clock {
  static async now(source: Source = 'external'): Promise<Date> {
    if (source === 'system') {
      return new Date();
    }

    const response = await fetch(`${env.CLOCK_URL}/clock?response_type=text`);
    const text = await response.text();
    return new Date(text);
  }
}
